features/nanostack/coap-service/source/coap_security_handler.c

```diff
@@ -396,7 +396,9 @@ int coap_security_handler_connect_non_blocking(coap_security_t *sec, bool is_ser
         mbedtls_ssl_conf_handshake_timeout(&sec->_conf, timeout_min, timeout_max);
     }
 
+#if !defined(MBEDTLS_SSL_CONF_RNG)
     mbedtls_ssl_conf_rng(&sec->_conf, mbedtls_ctr_drbg_random, &sec->_ctr_drbg);
+#endif
```

Comment:
Could we just delete this and always use the global RNG? What advantage is there in nanostack/coap having its own DRBG? I can currently only see downsides from a memory usage, code size, software maintainability, and security perspective.

Same for any other DRBG API use in the entire Mbed OS + Client stack. For best maintainability, security, and resource usages, we should always just be using one DRBG where possible, rather than making our use of the global DRBG dependent on a configuration option (which we have to test both ways).

Reply:
@jarvte Can you comment?

Reply:
Perhaps it would be acceptable to file a ticket against nanostack to move to using a single DRBG and handle that work in a separate PR.

Reply:
Antti and I thought that current solution is ok for this PR and we'll create a new task to take shared rng into use.

Reply:
Please comment with the ticket number when you have it.

Reply:
I just created a ticket IOTSSL-3000 for this.